# Practice: Sentiment Analysis

Now it's your turn to practice sentiment analysis!

## Sentiment Analyasis
In order to test out data mining on Bluesky, we are going to use a Natural Language Processing library, which gives us functions to work with langauge data such as sentences, words, etc.

You don't need to know the details of how these work internally, but we will be using the "[Natural Language Toolkit](https://www.nltk.org/)" along with the "Valence Aware Dictionary and sEntiment Reasoner" (or VADER) lexicon.

So let's get the library and lexicon now.

In [1]:
import nltk
nltk.download(["vader_lexicon"])
from nltk.sentiment import SentimentIntensityAnalyzer
sia = SentimentIntensityAnalyzer()

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\sw1918\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


Now that we have the "Sentiment Intensity Analyzer" we can try it out on different sentences to have the computer guess how positive or negative they are. Let's start with a really positive sentance:

In [2]:
sentence = "I love love love pizza!!!!!!!!"
sia.polarity_scores(sentence)["compound"]

0.941

In the above code, we ran a `polarity_scores` function on the sentence and pulled out the `compound` result. In this case it came back as `0.941`, which is close to 1 and indicates a positive statement.

Now let's try a really  negative one:

In [ ]:
sentence = "I hate hate hate pizza!!!!!!!!"
sia.polarity_scores(sentence)["compound"]re

Now, copy that two lines of code above, and try out your own sentences, and run the sentiment analysis on them. Try out several sentences!

In [ ]:
# TODO: enter your code here

### Looping with sentiments to find trends
If we want to look at trends over a series of sentences (or social media posts), we can use for loops.

So let's make an array of sentences and then do a loop to look at the sentiment of each sentence.

In [ ]:
# Make an array of movie reviews (each one a string)
movie_reviews = [
    "That movie was pretty good.",
    "I like the movie a lot!",
    "I thought the movie was boring",
    "I'd watch the movie again",
    "That sure was an exciting movie!"
]

Now, try to find the average sentiment score across all these movie reviews! 

Hint 1: You'll need to use a for loop
Hint 2: You'll need to create a variable outside the loop, to add up all the scores
Hint 3: You'll need to figure out the number of reviews in the list, in order to find the average

In [ ]:
# TODO: enter your code here